reports/tasks.py

```diff
@@ -0,0 +1,232 @@
+        'playbooks/collect_instance_statistics/collect_elasticsearch_data.yml'
+    )
+
+    def log_line(line):
+        """Helper to pass to capture_playbook_output()."""
+        sys.stdout.write(line)
+    log_line.info = log_line
+    log_line.error = log_line
+
+    # Launch the collect_elasticsearch_data playbook, which places a file into the `playbook_output_dir`
+    # on this host.
+    ansible.capture_playbook_output(
+        requirements_path=os.path.join(
+            os.path.dirname(playbook_path),
+            'requirements.txt'
+        ),
+        inventory_str=inventory,
+        vars_str=(
+            'local_output_dir: {output_dir}\n'
+            'remote_output_filename: /tmp/activity_report\n'
+            'server_name_prefix: {server_name_prefix}\n'
+            'start_date: {start_date}\n'
+            'end_date: {end_date}'
+            'config_section: {config_section}'
+        ).format(
+            output_dir=playbook_output_dir,
+            server_name_prefix=name_prefix,
+            start_date=start_date,
+            end_date=end_date,
+            config_section=name_prefix
+        ),
+        playbook_path=playbook_path,
+        username=settings.INSTANCE_LOGS_SERVER_SSH_USERNAME,
+        logger_=log_line,
+    )
+
+
+def get_instance_usage_data(playbook_output_dir, name_prefix, public_ip):
+    """ Execute the collect_activity playbook to gather statistics """
+    inventory = '[apps]\n{server}'.format(server=public_ip)
+    playbook_path = os.path.join(
+        settings.SITE_ROOT,
+        'playbooks/collect_activity/collect_activity.yml'
+    )
+
+    def log_line(line):
+        """Helper to pass to capture_playbook_output()."""
+        sys.stdout.write(line)
+    log_line.info = log_line
+    log_line.error = log_line
+
+    playbook_extra_script_arguments = '--skip-hit-statistics' if settings.INSTANCE_LOGS_SERVER_HOST else ''
+
+    # Launch the collect_activity playbook, which places a file into the `playbook_output_dir`
+    # on this host.
+    ansible.capture_playbook_output(
+        requirements_path=os.path.join(os.path.dirname(playbook_path), 'requirements.txt'),
+        inventory_str=inventory,
+        vars_str=(
+            'local_output_dir: {output_dir}\n'
+            'local_output_filename: user_statistics\n'
+            'remote_output_filename: /tmp/activity_report\n'
+            'config_section: {config_section}\n'
+            'extra_script_arguments: {extra_script_arguments}'
+        ).format(
+            output_dir=playbook_output_dir,
+            config_section=name_prefix,
+            extra_script_arguments=playbook_extra_script_arguments
+        ),
+        playbook_path=playbook_path,
+        username=settings.INSTANCE_LOGS_SERVER_SSH_USERNAME,
+        logger_=log_line,
+    )
+
+
+# Run on the 1st of every month
+@db_periodic_task(crontab(day='1', hour='2', minute='0'))
+def send_trial_instance_report(recipients=["patrick@opencraft.com"]):
```

Comment:
It would be nice to have the recipients added to the settings/env.

Reply:
Added this. Thanks for the suggestion